django/template/base.py

```diff
@@ -201,6 +202,24 @@ def render(self, context):
         finally:
             context.render_context.pop()
 
+    def _stream(self, context):
+        return self.nodelist.stream(context)
+
+    def stream(self, context):
+        "Display stage -- can be called many times"
+        context.render_context.push()
+        try:
+            if context.template is None:
+                with context.bind_template(self):
+                    context.template_name = self.name
+                    for chunk in self._stream(context):
+                        yield chunk
+            else:
+                for chunk in self._stream(context):
+                    yield chunk
+        finally:
+            context.render_context.pop()
```

Comment:
Could we add this in a context manager also? It's a finnicky code path I'd like to avoid duplicating if possible.

This worked for me in a quick test:

```python
class Template(object):
    
    @contextmanager
    def push_context(self, context):
        context.render_context.push()
        try:
            if context.template is None:
                with context.bind_template(self):
                    context.template_name = self.name
                    yield
            else:
                yield
        finally:
            context.render_context.pop()
    
    def render(self, context):
        with self.push_context(context):
            return self._render(context)
    
    def stream(self, context):
        with self.push_context(context):
            for chunk in self._stream(context):
                yield chunk
```